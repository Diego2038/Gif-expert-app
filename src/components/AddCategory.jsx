import { useState } from "react"

export const AddCategory = ( {hola, setCategories } ) => {

  const [inputValue, setInputValue] = useState('One punch');

  /**
   * FunciÃ³n encargada para tener una referencia entre
   * la variable reactiva y un elemento HTML
   * @param {ChangeEvent} event 
   */
  const onInputChange = ( event ) => {
    // console.log( event.target.value );
    setInputValue( event.target.value );
  } 

  const onSubmit = ( event ) => {
    event.preventDefault();
    if( inputValue.trim().length <= 1 ) return;
    // console.log( inputValue ); 
    setCategories( categories => [ inputValue, ...categories]);
    setInputValue( '' );
  }

  return (
    <form onSubmit={ onSubmit}>
      <input 
        type="text" 
        placeholder="Search a gif"
        value={ inputValue }
        onChange={onInputChange}   
      />
      <h6>{hola}</h6>
    </form>
  )
}
