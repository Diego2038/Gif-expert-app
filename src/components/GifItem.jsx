import { PropTypes} from 'prop-types';

export const GifItem = ( { id, url, title } ) => { 
  return (
    <div className="card">
      <img src={ url } alt={ title } aria-label="img-test" />
      <p>{ title }</p>
    </div>
  )
}


GifItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired
}
