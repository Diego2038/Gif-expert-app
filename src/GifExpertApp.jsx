import { useState } from "react";
import { AddCategory } from "./components/AddCategory";

export const GifExpertApp = () => {

  const [categories, setCategories] = useState(['One punch', 'DBZ']);

  const onNewCategory = ( value ) => {
    // console.log( value );
    setCategories( categories => [ value, ...categories ] );
  }

  /* const onAddCategory = () => {
    // setCategories( category => category.push('pepe'));
    console.log('hola')
    setCategories( ( cat ) => ['pepe', ...cat]);
  } */
  
  return (
    <>
    {/* TÃ­tulo */}
      <h1>GifExpertApp</h1>
    
    {/* Input */}
    <AddCategory 
    hola='mundo' 
    // setCategories={ setCategories } 
    onNewCategory={onNewCategory}
    /> 

    {/* Listado de Gifs */}
      {/* Gif item */}
    <ol> 
      { 
      categories.map( category => {
          return (<li key={ category }> {category} </li>)
        }) 
      }
      {/* <li>abc</li>  */}
    </ol>

    </>
  )
}
