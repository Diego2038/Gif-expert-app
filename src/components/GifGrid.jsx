import { GifItem } from "./GifItem"; 
import { useFetchGifs } from "../hooks/useFetchGifs";

export const GifGrid = ({ category }) => { 

  const { images, isLoading } = useFetchGifs( category );  
  
  return (
    <>
      <h3>{ category }</h3>

      {
        ( isLoading ) && 'Cargando...'
      }
       
      <div className="card-grid">
        {
          images.map( ( img ) =>
            <GifItem 
              key={ img.id } 
              {...img} />
          )
        }
      </div> 
    </>
  )
}
