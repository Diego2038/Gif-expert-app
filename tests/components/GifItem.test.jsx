import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components";

describe('test component GifItem', () => { 

  const title = 'Title # 1';
  const url = 'https://www.google.com/';

  test('should to match with the snapshot', () => { 
    const { container } = render( <GifItem key={'13'} title={title} url={url} />);

    expect( container ).toMatchSnapshot();
  });

  test('should to show the image with its respective alt and url', () => {
    render( <GifItem url={url} title={title} />);
    // screen.debug();
    const { src, alt } = screen.getByRole('img' );
    
    expect( src ).toBe( url );
    expect( alt ).toBe( title );
  });

  test('should to show the title in the component', () => {
    render( <GifItem url={url} title={title} /> );
    expect( screen.getByText( title ) ).toBeTruthy();
  })

});