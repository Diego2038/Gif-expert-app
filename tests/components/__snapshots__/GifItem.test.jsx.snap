// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test component GifItem should to match with the snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Title # 1"
      aria-label="img-test"
      src="https://www.google.com/"
    />
    <p>
      Title # 1
    </p>
  </div>
</div>
`;
